app/controllers/users/registrations_controller.rb

```diff
@@ -47,24 +47,20 @@ def process_successful_update(resource)
     end
 
     def process_updates(resource)
-      updater = UserFlashUpdater.new(resource, flash)
+      mobile_changed = @update_user_profile_form.mobile_changed?
+      updater = UserFlashUpdater.new(resource, flash, mobile_changed)
```

Comment:
I don't love the idea of passing in this third parameter to determine whether or not the mobile has changed. Instead, I propose that we pass in the form object instead of the user to `UserFlashUpdater`, and then we can call `#mobile_changed?` to find out. And for `needs_to_confirm_email_change?` to continue working, we would need to add `attr_reader :user` to the form, and then call `form.user` in `UserFlashUpdater`. I verified this works, but this means we have to rewrite the flash updater specs. I'd be happy to do this in a follow-up PR.